src/plugindata.h

```diff
@@ -263,8 +265,66 @@ void plugin_configure_single(GtkWidget *parent);
 void plugin_help(void);
 void plugin_cleanup(void);
 
+/** Called by Geany when a plugin library is loaded.
+ *
+ * This is the original entry point. Implement and export this function to be loadable at all.
+ * Then fill in GeanyPlugin::info and GeanyPlugin::funcs of the passed @p plugin. Finally
+ * GEANY_PLUGIN_REGISTER() and specify a minimum supported API version.
+ *
+ * For all glory details please read @ref howto.
+ *
+ * @param plugin The unique plugin handle to your plugin. You must set some fields here.
+ * @param module The GModule associated with your plugin library.
+ * @param geany_api_version The API version exposed by the Geany instance that loads your plugin.
+ *           This probably not the same as the one your plugin was compiled against.
```

Comment:
There are  some other possibilities like if a widget was added then ui_lookup_widget returns differently,
Plus, you can dlsym(NULL, "foobar") to dynamically load a function.

Both doesn't strictly require knowing the API version of Geany (since the function would just return NULL on older Geanys) but it makes it more soft-dependency more explizit and gives a hint as to whether soft-dependency code can be removed if the minimum supported API is bumped.

We could have geany_api_version() additionally. The param is for free basically. But I don't feel strong really, will remove it if you like.